carsus/io/cmfgen/base.py

```diff
@@ -464,3 +467,96 @@ def to_hdf(self, key='/photoionization_cross_sections'):
                     f.get_storer(subkey).attrs.metadata = self.base[i]._meta
 
                 f.root._v_attrs['metadata'] = self.meta
+
+
+class CMFGENReader:
+    """
+        Class for extracting levels and lines from CMFGEN.
+        
+        Mimics the GFALLReader class.
+
+        Attributes
+        ----------
+        levels : DataFrame
+        lines : DataFrame
+
+    """
+
+    def __init__(self, data, priority=10):
+        """
+        Parameters
+        ----------
+        data : dict
+            Dictionary containing one dictionary per specie with 
+            keys `levels` and `lines`.
+
+        priority: int, optional
+            Priority of the current data source, by default 10.
+        """
+        self.priority = priority
+        self._get_levels_lines(data)
+    
+    def _get_levels_lines(self, data):
+        """ Generates `levels` and `lines` DataFrames.
+
+        Parameters
+        ----------
+        data : dict
+            Dictionary containing one dictionary per specie with 
+            keys `levels` and `lines`.
+        """
+        lvl_list = []
+        lns_list = []
+        for ion, parser in data.items():
+
+            atomic_number = parse_selected_species(ion)[0][0]
+            ion_charge = parse_selected_species(ion)[0][1]
+            
+            lvl = parser['levels'].base
+            # some ID's have negative values (theoretical?)
+            lvl.loc[ lvl['ID'] < 0, 'method'] = 'theor'
+            lvl.loc[ lvl['ID'] > 0, 'method'] = 'meas'
+            lvl['ID'] = lvl['ID'].apply(np.abs)
+            lvl_id = lvl.set_index('ID')
+            lvl['atomic_number'] = atomic_number
+            lvl['ion_charge'] =  ion_charge
```

Comment:
`Si I` is (14, 0) so `ion_charge` is 0.

The `ion_charge` is later renamed to `ion_number` before output without any change. This happens also for GFALL and Chianti but seems a bit different for NIST (where `ion_number =  ion_charge +1`).

I did not implemented any of this, it was there before I arrived. Summarizing:

- All the readers have a column named `ion_charge` in their output DataFrame.
- Then, the `ion_charge` is renamed to `ion_number` in the `output` module.

Reply:
Thanks for the detailed explanation. We should make sure that this is consistent throughout all of the code. @wkerzendorf, @afloers and myself like ion_charge better than ion_number (since it is easier to interpret). Can you make an issue that we want to change to "ion_charge" also in the rest of the code?